docs/internal-documentation/templates/specifications/tests/Template01.md

```diff
@@ -0,0 +1,268 @@
+
+## Test procedure
+
+> > This section contains the detailed steps to execute the test case. The steps
+> > should be as explicit as possible to avoid that different implementations or
+> > executions do different interpretations or assumptions.
+> >
+> > The steps should be written in such a way that it is reasonably possible to
+> > use them to execute the test case manually using tools such as [`dig`]. It
+> > can be assumed that the reader of the text has good understanding of DNS.
+> >
+> > The steps should state what messages (message tags) to be outputted when.
+> > Only messages with default severity level DEBUG or higher can be included.
+> >
+> > All messages with level INFO or higher must be included. I.e. the
+> > implementation should not include messages with default level INFO or higher
+> > unless included in the specification.
+> >
+> > Messages DEBUG or lower can be added in the implementation as needed.
+> >
+> > Also include statement on what other information to be accompanied
+> > with the message (included as parameter to the message tag). Example of such
+> > information is IP adresses to name servers.
+> >
+> > When refering to the data defined in **Inputs** then use the name in
+> > *italic*, e.g.:
+
+2. Create a DNSKEY query with DO flag set for *Child Zone*.
+
+> > If data sets are created, then defined them in quote marks. E.g.:
+
+4. Create three empty sets, "NSEC-Zone", "NSEC3-Zone", and
+   "No-DNSSEC-Zone".
+
+> > When used (referred to) in the steps, make the name italic. E.g.:
+
+6. Add the name server IP to the *NSEC-Zone* set (*NSEC3-Zone*
+         set).
+
+> > When a message is outputted in the steps, it is done by outputting
+> > the message tag, "T01_HAS_NSEC3" in the example below. Make the tag as a link
+> > name in italic. What the link name points at is to be defined at the bottom
+> > of the document. E.g.:
+
+8. If the NSEC (NSEC3) records do not "cover" the
+   *Non-Existent Query Name*, then output *[T01_HAS_NSEC3]*
+
+## Outcome(s)
+> > First we have standard text that should normally be the same in all
+> > test case specifications.
+
+The outcome of this Test Case is "fail" if there is at least one message
+with the severity level *[ERROR]* or *[CRITICAL]*.
+
+The outcome of this Test Case is "warning" if there is at least one message
+with the severity level *[WARNING]*, but no message with severity level
+*ERROR* or *CRITICAL*.
+
+In other cases, no message or only messages with severity level
+*[INFO]* or *[NOTICE]*, the outcome of this Test Case is "pass".
+
+> > Then there could be statements about data being outputted from the test case
+> > for use as input data for other test cases.
+
+## Special procedural requirements
+
+> > First we have standard text that should normally be the same in all
+> > test case specifications.
+
+If either IPv4 or IPv6 transport is disabled, ignore the evaluation of the
+result of any test using this transport protocol. Log a message reporting
```

Comment:
`test` -> `request` maybe?

Reply:
I just copied the text from current specifications, but now I feel that it is wrong. When e.g. IPv6 is disabled the implementation does not ignore the evaluation, it skips running tests over that protocol. I will update here, but it will take a while before the specifications are updated.